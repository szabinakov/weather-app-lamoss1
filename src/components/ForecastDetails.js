import "../styles/ForecastDetails.css";
import "../styles/App.css";
import React from "react";
import getPokemonImageUrl from "../helpers/getPokemonImageUrl";

function ForecastDetails({ forecast }) {
  const { date, temperature, description, wind } = forecast;
  const pokemonImage = getPokemonImageUrl();
  // ...
  <img src={pokemonImage} alt="Pokemon" />;

  return (
    <div className="forecast-details-whole">
      <div className="forecast-details">
        <h2>{new Date(date).toLocaleDateString()}</h2>
        <div className="pokemon-background">
          <img src={pokemonImage} alt="Pokemon" />{" "}
        </div>
        <div className="forecast-details-main">
          <p>
            <strong>Temperature:</strong> {temperature.max}Â°C
          </p>
          <p>
            <strong>Description:</strong> {description}
          </p>
          <p>
            <strong>Wind Speed:</strong> {wind.speed} mph
          </p>
        </div>
      </div>
    </div>
  );
}

export default ForecastDetails;
