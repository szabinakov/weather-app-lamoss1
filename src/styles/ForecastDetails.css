.forecast-details h2 {
  font-size: 24px;
  font-weight: bold;
  margin-bottom: 10px;
  background-color: #fbfbfb;
}

.forecast-details-whole {
  background-color: #ccc;
}

.pokemon-background {
  background-color: #000;
}

.location-details p {
  font-size: 20px;
  margin-bottom: 10px;
  color: #666;
}

.forecast-details .pokemon-type {
  display: flex;
  flex-direction: row;
  font-size: 40px;
  margin-top: 10px;
  color: #e2001a;
}
