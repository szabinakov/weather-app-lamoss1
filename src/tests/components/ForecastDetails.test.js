import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    humidity: 30,
    wind: {
      speed: 100,
      direction: "stub",
    },
    temperature: {
      min: 12,
      max: 22,
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
