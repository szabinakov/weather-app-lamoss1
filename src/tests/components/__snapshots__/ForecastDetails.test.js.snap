// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details-whole"
  >
    <div
      class="forecast-details"
    >
      <h2>
        01/01/1970
      </h2>
      <div
        class="pokemon-background"
      >
        <img
          alt="Pokemon"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/24.png"
        />
         
      </div>
      <div
        class="forecast-details-main"
      >
        <p>
          <strong>
            Temperature:
          </strong>
           22Â°C
        </p>
        <p>
          <strong>
            Description:
          </strong>
           
        </p>
        <p>
          <strong>
            Wind Speed:
          </strong>
           100 mph
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
